### PR TITLE
Implement secure token service and blob URL core logic

### PR DESCRIPTION
**Pull request by engelxrg-25**
### Description
Brief description of the changes implemented in this PR.

### Related Issues
- Resolves feature/task-02
- Related to Epic #12

### Kiro Planning References
- [Requirements](.kiro/specs/blob-url-files/requirements.md)
- [Design](.kiro/specs/blob-url-files/design.md)
- [Tasks](.kiro/specs/blob-url-files/tasks.md)

### Changes Made
- [ ] Component implementation
- [ ] Service implementation
- [ ] Template updates
- [ ] Styling updates
- [ ] API integration
- [ ] Tests added

### Requirements Validation

- [ ] Requirement 1: User interface requirements met
- [ ] Requirement 2: Selection state requirements met
- [ ] Requirement 3: File operations requirements met
- [ ] Requirement 4: Permissions requirements met
- [ ] Requirement 5: Visual feedback requirements met
- [ ] Requirement 6: Progress tracking requirements met

### Testing Checklist
- [ ] Unit tests added/updated
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Edge cases tested
- [ ] Error handling tested

### Code Quality Checklist
- [ ] Code follows project standards
- [ ] TypeScript types properly defined
- [ ] Error handling implemented
- [ ] Loading states implemented
- [ ] No console.log statements left

### Review Checklist
- [ ] Functionality matches Kiro requirements
- [ ] Code follows Kiro design specifications
- [ ] Performance considerations addressed
- [ ] Security considerations addressed
- [ ] Accessibility requirements met

### Deployment Notes

---
*This PR was created as part of Kiro-planned feature development*

## Summary by CodeRabbit

- **New Features**
  - Temporary token-based download links with expiry, status, and richer metadata (filename, size, MIME type, creator, timestamps).
  - Secure token generation/validation and download retrieval flows; automatic cleanup of expired links.
- **Rate Limiting**
  - Per-IP, per-user, and token-validation quotas with enforced "Too many requests" behavior.
- **Validation**
  - Safer file-path validation to prevent traversal and unsafe inputs.
- **Tests**
  - New unit and integration tests covering end-to-end workflows.
- **Chores**
  - Added default rate-limit configuration and supporting dependencies.

### CODE REVIEW

File: src/main/java/com/sme/afs/service/BlobUrlService.java [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+                            "Rate limit exceeded for user. Please try again later.");
+                }
+                // Limit token validation attempts per IP to avoid brute force
+                String key = clientIp != null ? clientIp : "unknown";
+                if (!rateLimitService.isAllowed("token:validation:" + key)) {
+                    throw new AfsException(ErrorCode.TOO_MANY_REQUESTS,
+                            "Too many token validation attempts. Please try again later.");
+                }
+            }
+        } catch (AfsException e) {
```
[the remark this answers is not in the document]
Comment (engelxrg-25) [reply src/main/java/com/sme/afs/service/BlobUrlService.java:169]: @coderabbitai can you show me an example implementation?
